echo_client.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import unicode_literals
+import socket
+import sys
+
+
+def client_server(msg):
+    client_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP)
+
+    client_socket.connect(('127.0.0.1', 50000))
+    client_socket.sendall(msg.encode('utf-8'))
```

Comment:
What happens if the message passed in here to the client is a byte string containing non-ascii characters.  Write a test where the message you provide to the client is this: `b'\xc3\xbcbersetzer'` (you need the `b` since you are importing `unicode_literals`).  It will blow with a `UnicodeDecodeError` on this line.  

The problem is that in Python 2 you can call `encode` on a byte string (which is already encoded).  When you do, Python helpfully *decodes* the string first into a unicode object using `sys.defaultencoding`, which is usually `ASCII`.  that's not going to work for a word like übersetzer.